lib/galaxy/datatypes/proteomics.py

```diff
@@ -92,6 +233,277 @@ def display_peek(self, dataset):
         return self.make_html_table(dataset, column_names=self.column_names)
 
 
+@build_sniff_from_prefix
+class Dta(TabularData):
+    """dta
+    The first line contains the singly protonated peptide mass (MH+) and the
+    peptide charge state separated by a space. Subsequent lines contain space
+    separated pairs of fragment ion m/z and intensity values.
+    """
+    file_ext = "dta"
+    comment_lines = 0
+
+    def set_meta(self, dataset, **kwd):
+        column_types = []
+        data_row = []
+        data_lines = 0
+        if dataset.has_data():
+            with open(dataset.file_name, 'r') as dtafile:
```

Comment:
how much data is in such a file, can it be GB?

Reply:
I have no Idea -- but its a good idea to make this assumption. 

Maybe https://github.com/galaxyproject/galaxy/blob/80607164f4b33d4df0f0a7c2cc8f2e65a1b8e8ec/lib/galaxy/datatypes/data.py#L843 would be a solution. Since the `set_meta` function is of `TabularData` not implemented I can't use it via inheritance. 

Maybe it would be nice to implement this function as decorator or a utility function?

Note that csv / tsv types have the same problem https://github.com/galaxyproject/galaxy/blob/80607164f4b33d4df0f0a7c2cc8f2e65a1b8e8ec/lib/galaxy/datatypes/tabular.py#L940